torchgeo/datasets/south_africa_crop_type.py

```diff
@@ -0,0 +1,274 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""South Africa Crop Type Competition Dataset."""
+
+import os
+import re
+from collections.abc import Iterable
+from typing import Any, Callable, Optional, Union, cast
+
+import matplotlib.pyplot as plt
+import torch
+from matplotlib.figure import Figure
+from rasterio.crs import CRS
+from torch import Tensor
+
+from .geo import RasterDataset
+from .utils import BoundingBox, RGBBandsMissingError
+
+
+class SouthAfricaCropType(RasterDataset):
+    """South Africa Crop Type Challenge dataset.
+
+    The `South Africa Crop Type Challenge
+    <https://beta.source.coop/repositories/radiantearth/south-africa-crops-competition/description/>`__
+    dataset includes satellite imagery from Sentinel-1 and Sentinel-2 and labels for
+    crop type that were collected by aerial and vehicle survey from May 2017 to March
+    2018. Data was provided by the Western Cape Department of Agriculture and is
+    available via the Radiant Earth Foundation. Each chip is matched with a label.
+    Each pixel in the label contains an integer field number and crop type class.
```

Comment:
Maybe worth adding a note saying that the dataset contains time-series data, but that this is not yet exposed, and explain which time will be returned